dodola/core.py

```diff
@@ -40,3 +39,23 @@ def bias_correct_bcsd(
     predicted = model.predict(gcm_predict_ds[train_variable]).load()
     ds_predicted = predicted.to_dataset(name=out_variable)
     return ds_predicted
+
+
+def build_xesmf_weights_file(x, method, filename=None):
+    """Build ESMF weights file for regridding x to a global 1x1 grid
+
+    Parameters
+    ----------
+    x : xr.Dataset
+    method : str
+        Method of regridding. Passed to ``xesmf.Regridder``.
+    filename : optional
+        Local path to output netCDF weights file.
+
+    Returns
+    -------
+    outfilename : str
+        Path to resulting weights file.
+    """
+    out = xe.Regridder(x, xe.util.grid_global(1, 1), method=method, filename=filename)
```

Comment:
For creating the regridding file, I think we want to select out the first timestep of `x` because the regridder doesn't need all of `x` to create the weights file, so something like `out = xe.Regridder(x.isel(time=0), xe.util.grid_global(1, 1), method=method, filename=filename)` will speed this up. 

The other thing I'd suggest is adding a check for `method` in case the method passed isn't valid - so if it isn't `bilinear` or `conservative` we raise an error.

Reply:
On input checking: Unless there is a reason to perform the check on `method` here, I'd prefer to let `xe.Regridder` check and throw exceptions on bad input. We're pretty much just passing everything through so I feel like we don't have reason to perform the check in our code, because we don't need the check in our code (at least not right now). Do you know of a specific reason we should catch the input in `dodola` and not let it pass in?

On your point about selecting time. I don't think initializing `xe.Regridder` loads in data variables beyond `lon`, `lat`, `lon_b`, `lat_b` - which don't use `time` for dims so we don't get anything by slicing time?

Reply:
On input checking, that's fair - I don't know of a specific reason to catch a bad input in dodola versus just letting xesmf catch it. 

About time selection - you might be right? `xe.Regridder` _shouldn't_ need time -  and I don't think I've ever explicitly tested this. Let's leave as is for now.

Reply:
Yeah, I was hoping to do more comprehensive testing when we write the actual regridding service. We'll def have a good idea by then. Thanks again for checking this.